### ISSUE TITLE
Misaligned address error

### ISSUE THREAD

**Issue by lczyk** [alxyng/flappybird#2]
Having resolved #1, I am now getting `Job 1, 'build/flappybird' terminated by signal SIGBUS (Misaligned address error)`. After a bit of digging it looks like it's coming from line 1121 `texture = IMG_LoadTexture(renderer, "spritesheet.png");`.

I'm trying to run with `./build/Release/flappybird` so the sprite sheet should load fine.

I've also tried Debug build.

**Comment by lczyk** [comment]
Ok. More digging later and I've traced it down to, what i think might be a bug in SDL. Briefly, we're in `deps/SDL_Image/src/IMG_ImageIO.m` in `LoadImageFromFile`, we're making a CGIImageSource from file (this is fine) and then passing it to `CreateCGImageFromCGImageSource`. Here we're calling `image_ref = CGImageSourceCreateImageAtIndex(image_source, 0, NULL);` ([docs](https://developer.apple.com/documentation/imageio/cgimagesourcecreateimageatindex(_:_:_:))) and getting a SIGBUS error. If I change this line to `image_ref = CGImageSourceCreateImageAtIndex(image_source, -1, NULL);` aka getting an invalid image, it works fine.

I've added `printf("image_count: %lu\n", CGImageSourceGetCount(image_source));` and I'm getting `1`. Weird...
